components/collective-page/sections/Events.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { partition } from 'lodash';
+import memoizeOne from 'memoize-one';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import { isPastEvent } from '../../../lib/events';
+
+import { CONTRIBUTE_CARD_PADDING_X } from '../../contribute-cards/ContributeCardContainer';
+import ContributeCollective from '../../contribute-cards/ContributeCollective';
+import ContributeEvent from '../../contribute-cards/ContributeEvent';
+import CreateNew from '../../contribute-cards/CreateNew';
+import { Box, Flex } from '../../Grid';
+import HorizontalScroller from '../../HorizontalScroller';
+import Link from '../../Link';
+import StyledButton from '../../StyledButton';
+import { H3, P } from '../../Text';
+import ContainerSectionContent from '../ContainerSectionContent';
+import ContributeCardsContainer from '../ContributeCardsContainer';
+import SectionTitle from '../SectionTitle';
+
+class SectionEvents extends React.PureComponent {
+  static propTypes = {
+    /** Collective */
+    collective: PropTypes.shape({
+      slug: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      isActive: PropTypes.bool,
+    }).isRequired,
+    events: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        contributors: PropTypes.arrayOf(PropTypes.object),
+      }),
+    ),
+    connectedCollectives: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        collective: PropTypes.shape({
+          id: PropTypes.number.isRequired,
+        }),
+      }),
+    ),
+
+    isAdmin: PropTypes.bool.isRequired,
+  };
+
+  render() {
+    const { collective, events, connectedCollectives, isAdmin } = this.props;
+    const hasNoContributorForEvents = !events.find(event => event.contributors.length > 0);
+    const triageEvents = memoizeOne(events => {
```

Comment:
Since `triageEvents` will be re-declared on each render, memoization has no effect here.